realtek: Fix HW route lookup

oliv3r opened the pull request:
Fix various bugs found in the HW route lookup function.

Signed-off-by: Birger Koblitz <git@birger-koblitz.de>
Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

Commit:
realtek: Fix HW route lookup

Fix various bugs found in the HW route lookup function.

Signed-off-by: Birger Koblitz <git@birger-koblitz.de>
Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>

oliv3r commented:
@svanheule a patch in a series of 'small fixes' as individual MR's.

svanheule commented on target/linux/realtek/files-5.10/drivers/net/dsa/rtl83xx/rtl838x.h:
```diff
@@ -551,6 +551,14 @@ typedef enum {
+#define RTL930X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE_MASK       GENMASK(20, 19)
+#define RTL930X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE_OFFSET     19
+#define RTL903X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE(r) \
+        (((r) << RTL930X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE_OFFSET) & RTL930X_L3_HW_LU_KEY_CTRL_ROUTE_TYPE_MASK)
```
To make this more compact, you could use `FIELD_SET()` instead. Then you don't need the _OFFSET macro any more.

svanheule replied:
From bitfield.h, but apparently my memory failed me and it's called FIELD_PREP()
https://elixir.bootlin.com/linux/v6.1-rc2/source/include/linux/bitfield.h#L126

svanheule reviewed (commented):
The commit message mentions "various bugs", but I only see UC route lookup being fixed. If you know how the failing look-up manifests itself, this would be nice to info to have in the commit message.

I don't know how significantly you've reworked this patch, but these changes really look you've written them yourself. If you aren't indicating which changes you've made to the code, I think you should take authorship of this patch. Birger's contribution may then be more aptly registered by a Suggested-by, because I don't think he'll update this patch himself any more :-/

oliv3r commented:
> The commit message mentions "various bugs", but I only see UC route lookup being fixed. If you know how the failing look-up manifests itself, this would be nice to info to have in the commit message.
> 
I don't know. I think I actually split this patch already. This was the commit message given :(

> I don't know how significantly you've reworked this patch, but these changes really look you've written them yourself. If you aren't indicating which changes you've made to the code, I think you should take authorship of this patch. Birger's contribution may then be more aptly registered by a Suggested-by, because I don't think he'll update this patch himself any more :-/
I think this is a bigger discussion for later anyways. I haven't written this patch :) I've fixed some style issues and changed magic values for code.

I'm more then happy to drop my sign-off and keep it as birger's commit if that makes life easier :p I'm not too fussed as this entire driver needs to be re-written from mostly scratch anyway ...
